src/Microsoft.Health.Dicom.Core/Features/Store/StoreValidatorResultBuilder.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.ObjectModel;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class StoreValidatorResultBuilder
+{
+    public StoreValidatorResultBuilder()
+    {
+        ErrorMessages = new Collection<string>();
+        WarningMessages = new Collection<string>();
+        WarningCodes = ValidationWarnings.None;
+        FirstException = null;
+    }
+
+    private Collection<string> ErrorMessages { get; }
+
+    private Collection<string> WarningMessages { get; }
+
+    private ValidationWarnings WarningCodes { get; set; }
+
+    // TODO: Remove this during the cleanup. *** Hack to support the existing validator behavior ***
+    private Exception FirstException { get; set; }
+
+    public StoreValidatorResult Build()
+    {
+        return new StoreValidatorResult(
+            ErrorMessages,
+            WarningMessages,
+            WarningCodes,
+            FirstException);
+    }
+
+    public void AddError(Exception ex, QueryTag queryTag = null)
+    {
+        // TODO: Remove this during the cleanup. *** Hack to support the existing validator behavior ***
+        if (null == FirstException)
```

Comment:
nit: typically the variable is first in the comparison `FirstException == null` as we would say, "Is the 'FirstException' equal to null?"

Reply:
Done